### PR TITLE
KDESKTOP-1046-Reactivate-tests-on-Snapshot-class

### PR DESCRIPTION
**Pull request by ClementKunz**
Re-activate tests on Snapshot class

### CODE REVIEW

File: src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp [hunk 1]
```diff
@@ -69,16 +69,14 @@ bool Snapshot::updateItem(const SnapshotItem &newItem) {
     if (newItem.parentId().empty()) {
         LOG_WARN(Log::instance()->getLogger(), "Parent ID is empty for item " << newItem.id().c_str());
         assert(false);
-        return false;
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp:72]: For non-debug build types, the execution will continue without returning `false` immediately. Is it intended?

File: src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp [hunk 2]
```diff
@@ -69,16 +69,14 @@ bool Snapshot::updateItem(const SnapshotItem &newItem) {
...
     }
 
     if (newItem.parentId() == newItem.id()) {
         LOG_WARN(Log::instance()->getLogger(), "Parent ID equals item ID " << newItem.id().c_str());
         assert(false);
-        return false;
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp:78]: idem

File: src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp [hunk 3]
```diff
@@ -466,7 +447,6 @@ bool Snapshot::isOrphan(const NodeId &itemId) {
             // Should not happen
             LOG_WARN(Log::instance()->getLogger(), "Parent ID equals item ID " << nextParentId.c_str());
             assert(false);
-            break;
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp:469]: Same remark as above: the `break` will not be done for non-debug builds of the application.

File: test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp [hunk 4]
```diff
@@ -99,65 +50,73 @@ void TestSnapshot::tearDown() {
...
+    Snapshot snapshot(ReplicaSideLocal, dummyRootNode);
+
+    // Insert node A
+    const SnapshotItem itemA("a", SyncDb::driveRootNode().nodeIdLocal().value(), Str("A"), 1640995201, 1640995201,
+                             NodeType::NodeTypeDirectory, 123);
+    snapshot.updateItem(itemA);
+    CPPUNIT_ASSERT(snapshot.exists("a"));
+    CPPUNIT_ASSERT_EQUAL(std::string("A"), SyncName2Str(snapshot.name("a")));
+    CPPUNIT_ASSERT_EQUAL(NodeType::NodeTypeDirectory, snapshot.type("a"));
+    auto itItem = snapshot._items.find(SyncDb::driveRootNode().nodeIdLocal().value());
```
Comment (luc-guyot-infomaniak) [inline test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp:64]: Why do we need to access private attributes of the `Snapshot` class?

File: test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp [hunk 5]
```diff
@@ -99,65 +50,73 @@ void TestSnapshot::tearDown() {
...
-    _syncPal->_localSnapshot->updateItem(itemAAA);
-    CPPUNIT_ASSERT(_syncPal->_localSnapshot->exists("aaa"));
-    itItem = _syncPal->_localSnapshot->_items.find("aa");
-    CPPUNIT_ASSERT(itItem->second.childrenIds().find("aaa") != itItem->second.childrenIds().end());
+    const DbNode dummyRootNode(0, std::nullopt, SyncName(), SyncName(), "1", "1", std::nullopt, std::nullopt, std::nullopt,
+                               NodeTypeDirectory, 0, std::nullopt);
+    Snapshot snapshot(ReplicaSideLocal, dummyRootNode);
+
+    // Insert node A
+    const SnapshotItem itemA("a", SyncDb::driveRootNode().nodeIdLocal().value(), Str("A"), 1640995201, 1640995201,
```
Comment (luc-guyot-infomaniak) [inline test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp:58]: The value `SyncDb::driveRootNode().nodeIdLocal().value()` is used several times below.
Suggestion: defined a variable set with this value and re-use it.

File: test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp [hunk 6]
```diff
@@ -99,65 +50,73 @@ void TestSnapshot::tearDown() {
...
+    snapshot.updateItem(SnapshotItem("a", SyncDb::driveRootNode().nodeIdLocal().value(), Str("A*"), 1640995202, 1640995202,
+                                     NodeType::NodeTypeDirectory, 123));
+    CPPUNIT_ASSERT_EQUAL(std::string("A*"), SyncName2Str(snapshot.name("a")));
+
+    // Insert node B
+    const SnapshotItem itemB("b", SyncDb::driveRootNode().nodeIdLocal().value(), Str("B"), 1640995203, 1640995203,
+                             NodeType::NodeTypeDirectory, 123);
+    snapshot.updateItem(itemB);
+    CPPUNIT_ASSERT(snapshot.exists("b"));
+    itItem = snapshot._items.find(SyncDb::driveRootNode().nodeIdLocal().value());
```
Comment (luc-guyot-infomaniak) [inline test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp:77]: Same remark as above.

File: src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp (hunk 1 above)
Comment (ClementKunz) [reply src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp]: Done

File: src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp (hunk 3 above)
Comment (ClementKunz) [reply src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp]: Done

File: test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp (hunk 4 above)
Comment (ClementKunz) [reply test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp]: Done

File: test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp (hunk 5 above)
Comment (ClementKunz) [reply test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp]: Done

File: test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp (hunk 6 above)
Comment (ClementKunz) [reply test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp]: Done

**Review by luc-guyot-infomaniak — APPROVED** [review]

**Pull request merged**